### PR TITLE
feat: add extra headers to fetch of extraManifests

### PR DESCRIPTION
**Pull request by salkin**
Provides capability to add extra headers in cases where files can only be fetched with token based authenction.

Signed-off-by: Niklas Wik <niklas.wik@nokia.com>

### CODE REVIEW

File: pkg/config/types/v1alpha1/v1alpha1_types.go [hunk 1]
```diff
@@ -343,6 +343,14 @@ type ClusterConfig struct {
 	//         - "https://www.mysweethttpserver.com/manifest2.yaml"
 	ExtraManifests []string `yaml:"extraManifests,omitempty"`
 	//   description: |
+	//     A list of key:value pairs that will be added while fetching the ExtraManifests.
```
Comment (rsmitty) [inline pkg/config/types/v1alpha1/v1alpha1_types.go:346]: Would it make more sense to just have `Headers` be an optional part of a larger ExtraManifests struct? It seems likely that I would want headers for some, but not necessarily all, of the manifests I wish to deploy. I'm thinking something like this in the yaml:

```yaml
extraManifests:
  - url: http://test.com/file.yaml
    headers:
      - foo: bar
      - baz: test
  - url: http://test2.com/file2.yaml
```

**Comment by rsmitty** [comment]
Hey @salkin! Thanks for this. Looks like a good start, but I've got a question about the implementation in hopes we can make it a bit easier to use.

Also, you'll probably need to run `make docs` any time you make changes to the `v1alpha1_types.go` comments, since those become autogenerated docs and will fail CI if they don't match.

File: pkg/config/types/v1alpha1/v1alpha1_types.go (hunk 1 above)
Comment (andrewrynhard) [reply pkg/config/types/v1alpha1/v1alpha1_types.go:346]: Yeah I like this. It does make things a bit more verbose if each URL has the same headers, but we don't lose the ability to have different headers if we need to.

Comment (salkin) [reply pkg/config/types/v1alpha1/v1alpha1_types.go:346]: Are we ok with this as it would break v1alpha1 config?

Comment (andrewrynhard) [reply pkg/config/types/v1alpha1/v1alpha1_types.go:346]: Oh, I guess I didn't realize that it broke `v1alpha1` 🤔 . In that case, then I suppose the current implementation would have to work.

Comment (andrewrynhard) [reply pkg/config/types/v1alpha1/v1alpha1_types.go:346]: My only nit in that case, is that I think `ExtraManifestHeaders` should be a map. Then we don't have to split based on `:`.

**Comment by andrewrynhard** [comment]
@salkin Can you rebase? We had a chat about this today and decided that this should be fine for `v1alpha1`. Also, what do you think about making `ExtraManifestHeaders` a map?

**Comment by salkin** [comment]
Sure the map makes more change will redo to that

**Comment by salkin** [comment]
@andrewrynhard  updated to map and rebased, please check if ok.

**Comment by andrewrynhard** [comment]
@salkin Thanks! This looks good to me. I willl merge once we have 2 approvals.

**Comment by andrewrynhard** [comment]
@salkin Looks like you need to run `make docs`.

**Comment by andrewrynhard** [comment]
@salkin looks like linters got you. Try running `make lint-go`. We recently added a PR template that has guidance on how to test a PR locally, but this PR wasn’t created when we had it. Take a look in `.github`. It might help.

**Pull request merged**